internal/templates/nonexistentserviceaccount/template.go

```diff
@@ -0,0 +1,58 @@
+package nonexistentserviceaccount
+
+import (
+	"fmt"
+
+	"golang.stackrox.io/kube-linter/internal/check"
+	"golang.stackrox.io/kube-linter/internal/diagnostic"
+	"golang.stackrox.io/kube-linter/internal/extract"
+	"golang.stackrox.io/kube-linter/internal/lintcontext"
+	"golang.stackrox.io/kube-linter/internal/objectkinds"
+	"golang.stackrox.io/kube-linter/internal/set"
+	"golang.stackrox.io/kube-linter/internal/stringutils"
+	"golang.stackrox.io/kube-linter/internal/templates"
+	"golang.stackrox.io/kube-linter/internal/templates/nonexistentserviceaccount/internal/params"
+	v1 "k8s.io/api/core/v1"
+)
+
+var (
+	serviceAccountGVK = v1.SchemeGroupVersion.WithKind("ServiceAccount")
+)
+
+func init() {
+	templates.Register(check.Template{
+		HumanName:   "Non-Existent Service Account",
+		Key:         "non-existent-service-account",
+		Description: "Flag cases where a pod references a non-existent service account",
+		SupportedObjectKinds: check.ObjectKindsDesc{
+			ObjectKinds: []string{objectkinds.DeploymentLike},
+		},
+		Parameters:             params.ParamDescs,
+		ParseAndValidateParams: params.ParseAndValidate,
+		Instantiate: params.WrapInstantiateFunc(func(_ params.Params) (check.Func, error) {
+			return func(lintCtx *lintcontext.LintContext, object lintcontext.Object) []diagnostic.Diagnostic {
+				podSpec, found := extract.PodSpec(object.K8sObject)
+				if !found {
+					return nil
+				}
+				sa := stringutils.OrDefault(podSpec.ServiceAccountName, podSpec.DeprecatedServiceAccount)
+				if sa == "" {
+					return nil
+				}
+				ns := object.K8sObject.GetNamespace()
+				serviceAccountsInCtx := set.NewStringSet()
+				for _, otherObj := range lintCtx.Objects() {
```

Comment:
Yeah, I had a similar thought. I started working on this on another branch -- I think we need a more generic solution because I expect we'll want to cache other things. Are you okay if I defer this to the follow-up PR?

Reply:
Yeah that's fine. Having a more generic interface definitely makes sense